### PR TITLE
Note changes in unicode escapes and case companion

### PR DESCRIPTION
**Pull request by som-snytt**
Scala 3 acquired extra help for case companions which was backported to Scala 2 (under `-Xsource:3-cross`). 🤞 

I did not build locally to review output, so this is best-effort for formatting.

### CODE REVIEW

**Comment by som-snytt** [comment]
It looks like CI fails for SIP links to virtus labs. Difficult to keep internal docs in sync with external sites that may evolve dynamically.

File: _overviews/scala3-migration/incompat-syntactic.md [hunk 1]
```diff
@@ -237,3 +238,13 @@ The solution is to choose another valid identifier, for instance `T`.
 
 However, `+` and `-` still are valid type identifiers in general.
 You can write `type +`.
+
+## Unicode escapes
+
+Unicode escapes of the form `\u0061` are no longer natively supported in any position by the compiler.
```
Comment (SethTisue) [inline _overviews/scala3-migration/incompat-syntactic.md:244]: "natively supported" seems unclear to me. not sure what best replacement is, maybe:

```suggestion
Unicode escapes of the form `\u0061` are no longer allowed in arbitrary locations in source code.
```

File: _overviews/scala3-migration/incompat-syntactic.md [hunk 2]
```diff
@@ -237,3 +238,13 @@ The solution is to choose another valid identifier, for instance `T`.
...
+
+Instead, they processed like other escapes in string literals.
```
Comment (SethTisue) [inline _overviews/scala3-migration/incompat-syntactic.md:246]:
```suggestion
Instead, they are processed in string literals, just like other escape sequences.
```

File: _overviews/scala3-migration/incompat-syntactic.md [hunk 3]
```diff
@@ -237,3 +238,13 @@ The solution is to choose another valid identifier, for instance `T`.
...
+This means, in particular, that they are not processed in "triple-quoted" strings or by the `raw` interpolator.
+
+In Scala 2, a deprecation is issued, but under `-Xsource:3cross`, the behavior is the same as Scala 3.
```
Comment (SethTisue) [inline _overviews/scala3-migration/incompat-syntactic.md:249]:
```suggestion
In Scala 2, a deprecation is issued, but under `-Xsource:3-cross`, the behavior is the same as Scala 3.
```

**Pull request closed without merging**